From 4833f02e47d5b7fc4284517e92d5976b20188a4a Mon Sep 17 00:00:00 2001
From: Arun Raghavan <arun@arunraghavan.net>
Date: Thu, 8 Sep 2016 17:30:41 +0530
Subject: [PATCH] pulsesrc: Don't negotiate to less than two segments

GstAudioRingBuffer doesn't needs us to have at least 2 segments. We make
sure that if our buffer parameters are such that the maxlength is not at
least 2x fragsize, we still request the ringbuffer to keep that much
space so it continues to work.

https://bugzilla.gnome.org/show_bug.cgi?id=770446
---
 ext/pulse/pulsesrc.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/ext/pulse/pulsesrc.c b/ext/pulse/pulsesrc.c
index 6ee2c506d..7af74ee6e 100644
--- a/ext/pulse/pulsesrc.c
+++ b/ext/pulse/pulsesrc.c
@@ -1595,6 +1595,9 @@ gst_pulsesrc_prepare (GstAudioSrc * asrc, GstAudioRingBufferSpec * spec)
 
   /* Fix up the total ringbuffer size based on what we actually got */
   spec->segtotal = actual->maxlength / spec->segsize;
+  /* Don't buffer less than 2 segments as the ringbuffer can't deal with it */
+  if (spec->segtotal < 2)
+    spec->segtotal = 2;
 
   if (!pulsesrc->paused) {
     GST_DEBUG_OBJECT (pulsesrc, "uncorking because we are playing");
